wandb/sdk/wandb_run.py

```diff
@@ -378,19 +378,30 @@ class Run:
     _is_attached: bool
     _settings: Settings
 
+    _launch_artifacts: Optional[Dict[str, Any]]
+
     def __init__(
         self,
         settings: Settings,
         config: Optional[Dict[str, Any]] = None,
+        sweep_config: Optional[Dict[str, Any]] = None,
+        launch_config: Optional[Dict[str, Any]] = None,
     ) -> None:
         # pid is set, so we know if this run object was initialized by this process
         self._init_pid = os.getpid()
-        self._init(settings=settings, config=config)
+        self._init(
+            settings=settings,
+            config=config,
+            sweep_config=sweep_config,
+            launch_config=launch_config,
+        )
 
     def _init(
         self,
         settings: Settings,
         config: Optional[Dict[str, Any]] = None,
+        sweep_config: Optional[Dict[str, Any]] = None,
+        launch_config: Optional[Dict[str, Any]] = None,
```

Comment:
do we need to add new arguments here?
can we just have a helper method on the Run that updates the config ?
like `Run._update_config(config, user)`

Reply:
This PR starts off by reverting the changes from [this PR](https://github.com/wandb/client/pull/3582), returning the sweep_config as an argument so that it can be used with the UpsertRun. In order to make merging sweeps and launch simpler for the future I'm handling the launch config in the same way. That way we at least have a consistent manner of passing these configs in the upsert mutation.


The reason we're modifying this change is because of the config debounce causing a delay between the shown config values in the UI which has been confusing some users.

Reply:
I see... I guess I'm missing something... just trying to understand why we need to have multiple config args? are they updating the same final config? can we just pass whatever config we have to some helper function and updated?
what is the advantaged to have it explicitly specified?

Reply:
The sweep_config and launch_config each separately update the config and lock the related keys https://github.com/wandb/client/pull/3734/files#diff-b63db14fd18e0e4d240ec68e625f5420067376cb574e03f383ca7a29e7bfce4bR503

Then the ini_config is applied, and then the run is upserted with the resulting config.

Keeping them separate lets the user know what locked what in case both of these things are ever set. Although right now we wouldn't expect that. I think keeping these two configs separate will make it easier to maintain. Thoughts?